### ISSUE TITLE
Exception when adding externalId and changing id for an existing group

### ISSUE THREAD

**Issue by phornig** [Netcentric/accesscontroltool#575]
With 2.8.0 I get an exception when trying to add an externalId when also changing the id (changing some letters to upper case) of an existing group :

```
29.04.2021 14:53:01.948 *INFO* [OsgiInstallerImpl] biz.netcentric.cq.tools.actool.installhook.impl.OsgiAwareInstallHook Exception while installing configurations: biz.netcentric.cq.tools.actool.authorizableinstaller.AuthorizableCreatorException: java.lang.Ille
galStateException: Authorizable content-editor: underlying tree has been disconnected.
29.04.2021 14:53:01.948 *WARN* [OsgiInstallerImpl] org.apache.jackrabbit.vault.packaging.impl.InstallHookProcessorImpl Hook actool threw package exception. PREPARE aborted.
org.apache.jackrabbit.vault.packaging.PackageException: java.lang.IllegalStateException: Authorizable content-editor: underlying tree has been disconnected.
        at biz.netcentric.cq.tools.actool.installhook.AcToolInstallHook.install(AcToolInstallHook.java:106) [biz.netcentric.cq.tools.accesscontroltool.bundle:2.8.0]
        at biz.netcentric.cq.tools.actool.installhook.AcToolInstallHook.execute(AcToolInstallHook.java:42) [biz.netcentric.cq.tools.accesscontroltool.bundle:2.8.0]
        at org.apache.jackrabbit.vault.packaging.impl.InstallHookProcessorImpl.execute(InstallHookProcessorImpl.java:143) [org.apache.jackrabbit.vault:3.4.0]
        at org.apache.jackrabbit.vault.packaging.impl.ZipVaultPackage.extract(ZipVaultPackage.java:226) [org.apache.jackrabbit.vault:3.4.0]
        at org.apache.jackrabbit.vault.packaging.impl.JcrPackageImpl.extract(JcrPackageImpl.java:401) [org.apache.jackrabbit.vault:3.4.0]
        at org.apache.jackrabbit.vault.packaging.impl.JcrPackageImpl.extract(JcrPackageImpl.java:360) [org.apache.jackrabbit.vault:3.4.0]
        at org.apache.jackrabbit.vault.packaging.impl.JcrPackageImpl.install(JcrPackageImpl.java:354) [org.apache.jackrabbit.vault:3.4.0]
        at com.adobe.granite.installer.factory.packages.impl.PackageTransformer$InstallPackageTask.execute(PackageTransformer.java:350) [com.adobe.granite.installer.factory.packages:1.0.18]
        at org.apache.sling.installer.core.impl.OsgiInstallerImpl.doExecuteTasks(OsgiInstallerImpl.java:918) [org.apache.sling.installer.core:3.11.4]
        at org.apache.sling.installer.core.impl.OsgiInstallerImpl.executeTasks(OsgiInstallerImpl.java:755) [org.apache.sling.installer.core:3.11.4]
        at org.apache.sling.installer.core.impl.OsgiInstallerImpl.run(OsgiInstallerImpl.java:304) [org.apache.sling.installer.core:3.11.4]
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: biz.netcentric.cq.tools.actool.authorizableinstaller.AuthorizableCreatorException: java.lang.IllegalStateException: Authorizable content-editor: underlying tree has been disconnected.
        at biz.netcentric.cq.tools.actool.impl.AcInstallationServiceImpl.installAuthorizables(AcInstallationServiceImpl.java:564) [biz.netcentric.cq.tools.accesscontroltool.bundle:2.8.0]
        at biz.netcentric.cq.tools.actool.impl.AcInstallationServiceImpl.installAcConfiguration(AcInstallationServiceImpl.java:343) [biz.netcentric.cq.tools.accesscontroltool.bundle:2.8.0]
        at biz.netcentric.cq.tools.actool.impl.AcInstallationServiceImpl.installMergedConfigurations(AcInstallationServiceImpl.java:668) [biz.netcentric.cq.tools.accesscontroltool.bundle:2.8.0]
        at biz.netcentric.cq.tools.actool.impl.AcInstallationServiceImpl.installConfigurationFiles(AcInstallationServiceImpl.java:304) [biz.netcentric.cq.tools.accesscontroltool.bundle:2.8.0]
        at biz.netcentric.cq.tools.actool.impl.AcInstallationServiceImpl.installConfigurationFiles(AcInstallationServiceImpl.java:270) [biz.netcentric.cq.tools.accesscontroltool.bundle:2.8.0]
        at biz.netcentric.cq.tools.actool.installhook.impl.AcToolInstallHookServiceImpl.installYamlFilesFromPackage(AcToolInstallHookServiceImpl.java:63) [biz.netcentric.cq.tools.accesscontroltool.bundle:2.8.0]
        at biz.netcentric.cq.tools.actool.installhook.AcToolInstallHook.install(AcToolInstallHook.java:99) [biz.netcentric.cq.tools.accesscontroltool.bundle:2.8.0]
        ... 11 common frames omitted
Caused by: java.lang.IllegalStateException: Authorizable content-editor: underlying tree has been disconnected.
        at org.apache.jackrabbit.oak.security.user.AuthorizableImpl.getTree(AuthorizableImpl.java:191) [org.apache.jackrabbit.oak-core:1.39.0.R1887543]
        at org.apache.jackrabbit.oak.security.user.AuthorizableImpl.getMembership(AuthorizableImpl.java:285) [org.apache.jackrabbit.oak-core:1.39.0.R1887543]
        at org.apache.jackrabbit.oak.security.user.AuthorizableImpl.memberOfMonitored(AuthorizableImpl.java:263) [org.apache.jackrabbit.oak-core:1.39.0.R1887543]
        at org.apache.jackrabbit.oak.security.user.AuthorizableImpl.declaredMemberOf(AuthorizableImpl.java:88) [org.apache.jackrabbit.oak-core:1.39.0.R1887543]
        at org.apache.jackrabbit.oak.jcr.delegate.AuthorizableDelegator$4.perform(AuthorizableDelegator.java:117) [org.apache.jackrabbit.oak-jcr:1.39.0.R1887543]
        at org.apache.jackrabbit.oak.jcr.delegate.AuthorizableDelegator$4.perform(AuthorizableDelegator.java:113) [org.apache.jackrabbit.oak-jcr:1.39.0.R1887543]
        at org.apache.jackrabbit.oak.jcr.delegate.SessionDelegate.perform(SessionDelegate.java:209) [org.apache.jackrabbit.oak-jcr:1.39.0.R1887543]
        at org.apache.jackrabbit.oak.jcr.delegate.AuthorizableDelegator.declaredMemberOf(AuthorizableDelegator.java:113) [org.apache.jackrabbit.oak-jcr:1.39.0.R1887543]
        at biz.netcentric.cq.tools.actool.authorizableinstaller.impl.AuthInstallerUserManagerPrefetchingImpl.getDeclaredIsMemberOf(AuthInstallerUserManagerPrefetchingImpl.java:120) [biz.netcentric.cq.tools.accesscontroltool.bundle:2.8.0]
        at biz.netcentric.cq.tools.actool.authorizableinstaller.impl.AuthorizableInstallerServiceImpl.applyGroupMembershipConfigIsMemberOf(AuthorizableInstallerServiceImpl.java:513) [biz.netcentric.cq.tools.accesscontroltool.bundle:2.8.0]
        at biz.netcentric.cq.tools.actool.authorizableinstaller.impl.AuthorizableInstallerServiceImpl.installAuthorizableConfigurationBean(AuthorizableInstallerServiceImpl.java:170) [biz.netcentric.cq.tools.accesscontroltool.bundle:2.8.0]
        at biz.netcentric.cq.tools.actool.authorizableinstaller.impl.AuthorizableInstallerServiceImpl.installAuthorizables(AuthorizableInstallerServiceImpl.java:119) [biz.netcentric.cq.tools.accesscontroltool.bundle:2.8.0]
        at biz.netcentric.cq.tools.actool.impl.AcInstallationServiceImpl.installAuthorizables(AcInstallationServiceImpl.java:551) [biz.netcentric.cq.tools.accesscontroltool.bundle:2.8.0]
        ... 17 common frames omitted
29.04.2021 14:53:01.949 *ERROR* [OsgiInstallerImpl] com.adobe.granite.installer.factory.packages.impl.PackageTransformer Error while processing install task of Abbott:aem-platform-cloud.permissions:1.0.0-SNAPSHOT.
org.apache.jackrabbit.vault.packaging.PackageException: Error while executing an install hook during prepare phase.
        at org.apache.jackrabbit.vault.packaging.impl.ZipVaultPackage.extract(ZipVaultPackage.java:229) [org.apache.jackrabbit.vault:3.4.0]
        at org.apache.jackrabbit.vault.packaging.impl.JcrPackageImpl.extract(JcrPackageImpl.java:401) [org.apache.jackrabbit.vault:3.4.0]
        at org.apache.jackrabbit.vault.packaging.impl.JcrPackageImpl.extract(JcrPackageImpl.java:360) [org.apache.jackrabbit.vault:3.4.0]
        at org.apache.jackrabbit.vault.packaging.impl.JcrPackageImpl.install(JcrPackageImpl.java:354) [org.apache.jackrabbit.vault:3.4.0]
        at com.adobe.granite.installer.factory.packages.impl.PackageTransformer$InstallPackageTask.execute(PackageTransformer.java:350) [com.adobe.granite.installer.factory.packages:1.0.18]
        at org.apache.sling.installer.core.impl.OsgiInstallerImpl.doExecuteTasks(OsgiInstallerImpl.java:918) [org.apache.sling.installer.core:3.11.4]
        at org.apache.sling.installer.core.impl.OsgiInstallerImpl.executeTasks(OsgiInstallerImpl.java:755) [org.apache.sling.installer.core:3.11.4]
        at org.apache.sling.installer.core.impl.OsgiInstallerImpl.run(OsgiInstallerImpl.java:304) [org.apache.sling.installer.core:3.11.4]
        at java.base/java.lang.Thread.run(Thread.java:834)
```

Steps to reproduce:
1. Create a group
```- group_config:

    - test-group:
        - name: Test Group
          path: /home/groups/test
```
2. Change group name and add external id
```- group_config:

    - TEST-group:
        - name: Test Group
          path: /home/groups/test
          externalId: "TEST-group"
```

It looks like this is related to https://github.com/Netcentric/accesscontroltool/issues/542 and https://github.com/Netcentric/accesscontroltool/issues/562.

**Comment by phornig** [comment]
The root cause seems to be different handling of the authorizable id in the UserManager delegate and the authorizable cache within AuthInstallerUserManagerPrefetchingImpl. So always using lower case for the key when adding or retrieving from the authorizable cache aligns the handling and solves this issue.

**Comment by ghenzler** [comment]
Merged #576
